### PR TITLE
Mutable async context example

### PR DESCRIPTION
**Pull request by esseswann**

### CODE REVIEW

File: docs/book/content/types/objects/using_contexts.md [hunk 1]
```diff
@@ -87,4 +86,55 @@ impl User {
...
+If you are using async runtime like `tokio` for mutable references you will need to use a corresponding async version of RwLock:
+```rust
+# extern crate juniper;
+# use std::collections::HashMap;
+# use juniper::graphql_object;
+# use tokio::sync::RwLock;
+#
+// This struct represents our context.
+struct Database {
+    requested_count: HashMap<i32, i32>,
```
Comment (kpp) [inline docs/book/content/types/objects/using_contexts.md:99]: Why HashMap?

Comment (esseswann) [reply docs/book/content/types/objects/using_contexts.md:99]: It was cloned from previous example, what do you think is more appropriate here?

Comment (kpp) [reply docs/book/content/types/objects/using_contexts.md:99]: I was mislead by `.entry(0)`.

**Review by tyranron — CHANGES_REQUESTED** [review]
@esseswann thanks 🍻

But CI errors seem to be legit. Please, update the provided example so it passes the Book tests.

**Comment by LegNeato** [comment]
Thank you!

**Pull request merged**